Snapshots fail with NullPointerException on Linux ARM64

### CrateDB version

6.3.5

### CrateDB setup information

Architecture: Linux ARM64
Number of nodes: any
CRATE_HEAP_SIZE: any
CRATE_JAVA_OPTS: 

crate.yml contents:

### Problem description

Creating new snapshots and creating new repositories fails across all snapshot filesystems.

Seems like this wasn't fully resolved: https://github.com/crate/crate/issues/19586

I believe the bug is in OpenDAL itself: https://github.com/apache/opendal/issues/7905

### Steps to Reproduce

1. CREATE REPOSITORY testrepo TYPE s3 WITH (bucket='a-bucket-with-write-access', access_key='AKIA.....', secret_key='dirty-little-secret', endpoint='s3.dualstack.us-east-2.amazonaws.com', region='us-east-2', compress=true);

### Actual Result

```
RepositoryException[[] [testrepo] Unable to verify the repository, [testrepo] is not accessible on master node: NullPointerException 'null']

org.elasticsearch.repositories.RepositoryException: [] [testrepo] Unable to verify the repository, [testrepo] is not accessible on master node: NullPointerException 'null'
	at io.crate.execution.ddl.RepositoryService.convertRepositoryException(RepositoryService.java:128)
	at io.crate.execution.ddl.RepositoryService.lambda$execute$2(RepositoryService.java:87)
	at java.base/java.util.concurrent.CompletableFuture$UniComposeExceptionally.tryFire(CompletableFuture.java:1061)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:531)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2221)
	at io.crate.concurrent.FutureActionListener.onFailure(FutureActionListener.java:37)
	at org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.lambda$doStart$1(TransportMasterNodeAction.java:173)
	at org.elasticsearch.action.ActionListener.lambda$withOnFailure$0(ActionListener.java:125)
	at org.elasticsearch.action.ActionListener$SimpleListener.onFailure(ActionListener.java:106)
	at org.elasticsearch.action.ActionListener.accept(ActionListener.java:55)
	at org.elasticsearch.action.ActionListener.accept(ActionListener.java:38)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:884)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:862)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:531)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2221)
	at io.crate.concurrent.FutureActionListener.onFailure(FutureActionListener.java:37)
	at org.elasticsearch.action.ActionRunnable.onFailure(ActionRunnable.java:90)
	at org.elasticsearch.common.util.concurrent.RejectableRunnable.run(RejectableRunnable.java:38)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base/java.lang.Thread.run(Thread.java:1474)
Caused by: org.elasticsearch.repositories.RepositoryVerificationException: [testrepo] Unable to verify the repository, [testrepo] is not accessible on master node: NullPointerException 'null'
	at org.elasticsearch.repositories.blobstore.BlobStoreRepository.startVerification(BlobStoreRepository.java:1210)
	at org.elasticsearch.repositories.RepositoriesService$3.doRun(RepositoriesService.java:252)
	at org.elasticsearch.common.util.concurrent.RejectableRunnable.run(RejectableRunnable.java:36)
	... 3 more
Caused by: java.lang.NullPointerException
	at org.apache.opendal.OperatorOutputStream.writeBytes(Native Method)
	at org.apache.opendal.OperatorOutputStream.flush(OperatorOutputStream.java:73)
	at org.apache.opendal.OperatorOutputStream.write(OperatorOutputStream.java:61)
	at java.base/java.io.OutputStream.write(OutputStream.java:167)
	at java.base/java.io.InputStream.transferTo(InputStream.java:796)
	at io.crate.opendal.OpenDALBlobContainer.writeBlob(OpenDALBlobContainer.java:86)
	at org.elasticsearch.common.blobstore.BlobContainer.writeBlobAtomic(BlobContainer.java:119)
	at org.elasticsearch.repositories.blobstore.BlobStoreRepository.startVerification(BlobStoreRepository.java:1201)
	... 5 more
	Suppressed: java.lang.NullPointerException
		at org.apache.opendal.OperatorOutputStream.writeBytes(Native Method)
		at org.apache.opendal.OperatorOutputStream.flush(OperatorOutputStream.java:73)
		at org.apache.opendal.OperatorOutputStream.close(OperatorOutputStream.java:80)
		at io.crate.opendal.OpenDALBlobContainer.writeBlob(OpenDALBlobContainer.java:85)
		... 7 more
```

### Expected Result

Repository creation succeeds.

--- a/3RD-PARTY-NOTICES.md
+++ b/3RD-PARTY-NOTICES.md
@@ -68,7 +68,7 @@ Generated using [License Maven Plugin](https://www.mojohaus.org/license-maven-pl
 - (Apache 2) Apache Lucene (module: spatial3d) (org.apache.lucene:lucene-spatial3d:10.3.2 - https://lucene.apache.org/)
 - (Apache 2) Apache Lucene (module: suggest) (org.apache.lucene:lucene-suggest:10.3.2 - https://lucene.apache.org/)
 - (Apache 2) Apache Lucene (module: test-framework) (org.apache.lucene:lucene-test-framework:10.3.2 - https://lucene.apache.org/)
-- (Apache-2.0) Apache OpenDAL™ (org.apache.opendal:opendal:0.48.2 - https://opendal.apache.org)
+- (Apache-2.0) Apache OpenDAL™ (org.apache.opendal:opendal:0.50.1 - https://opendal.apache.org)
 - (The Apache License, Version 2.0) org.apiguardian:apiguardian-api (org.apiguardian:apiguardian-api:1.1.2 - https://github.com/apiguardian-team/apiguardian)
 - (Apache-2.0) AssertJ Core (org.assertj:assertj-core:3.27.6 - https://assertj.github.io/doc/#assertj-core)
 - (The MIT License) Checker Qual (org.checkerframework:checker-qual:3.49.5 - https://checkerframework.org/)

--- a/docs/appendices/release-notes/6.3.7.rst
+++ b/docs/appendices/release-notes/6.3.7.rst
@@ -86,3 +86,8 @@ Fixes
   ``variance``, a negative number) instead of ``0`` when aggregating
   large-magnitude values with little or no spread, such as a constant ``DATE``
   or ``TIMESTAMP``.
+
+- Fixed an issue that led to ``NullPointerException`` being thrown when
+  :ref:`creating an S3 repository <sql-create-repo-s3>` on Linux ARM64. Special
+  thanks to `Robert Palmer <https://github.com/robd003>`_ for fixing the issue
+  on the upstream project.

--- a/docs/appendices/release-notes/6.4.2.rst
+++ b/docs/appendices/release-notes/6.4.2.rst
@@ -86,3 +86,8 @@ Fixes
   ``variance``, a negative number) instead of ``0`` when aggregating
   large-magnitude values with little or no spread, such as a constant ``DATE``
   or ``TIMESTAMP``.
+
+- Fixed an issue that led to ``NullPointerException`` being thrown when
+  :ref:`creating an S3 repository <sql-create-repo-s3>` on Linux ARM64. Special
+  thanks to `Robert Palmer <https://github.com/robd003>`_ for fixing the issue
+  on the upstream project.

--- a/plugins/cr8-copy-s3/src/main/java/io/crate/copy/s3/common/S3Env.java
+++ b/plugins/cr8-copy-s3/src/main/java/io/crate/copy/s3/common/S3Env.java
@@ -23,6 +23,7 @@ package io.crate.copy.s3.common;
 
 import org.apache.opendal.ServiceConfig;
 import org.elasticsearch.common.settings.Settings;
+import org.jspecify.annotations.Nullable;
 
 import io.crate.opendal.S3;
 
@@ -35,15 +36,19 @@ public final class S3Env {
         String protocol = S3Protocol.get(withClause);
         String endpoint = protocol + "://" + s3uri.endpoint();
         String region = S3.getRegion(endpoint, s3uri.bucket());
-        return ServiceConfig.S3.builder()
+        @Nullable
+        String accessKey = s3uri.accessKey();
+        var builder = ServiceConfig.S3.builder()
             .bucket(s3uri.bucket())
             .endpoint(endpoint)
-            .accessKeyId(s3uri.accessKey())
+            .accessKeyId(accessKey)
             .secretAccessKey(s3uri.secretKey())
             .disableConfigLoad(true)
             .disableEc2Metadata(true)
-            .allowAnonymous(true)
-            .region(region)
-            .build();
+            .region(region);
+        if (accessKey == null) {
+            builder.skipSignature(true);
+        }
+        return builder.build();
     }
 }

--- a/pom.xml
+++ b/pom.xml
@@ -342,7 +342,7 @@
         <versions.caffeine>3.2.4</versions.caffeine>
         <versions.jodatime>2.14.2</versions.jodatime>
         <versions.aws>1.12.797</versions.aws>
-        <versions.opendal>0.48.2</versions.opendal>
+        <versions.opendal>0.50.1</versions.opendal>
         <versions.commonsmath>3.6.1</versions.commonsmath>
         <versions.commonscodec>1.22.0</versions.commonscodec>
         <versions.bigmath>2.3.2</versions.bigmath>

